plugin/trino-prometheus/src/test/java/io/trino/plugin/prometheus/TestPrometheusIntegration.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.prometheus;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.Duration;
+import io.trino.spi.connector.ConnectorSplitSource;
+import io.trino.spi.connector.DynamicFilter;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.MaterializedResult;
+import io.trino.testing.MaterializedRow;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static io.trino.plugin.prometheus.PrometheusQueryRunner.createPrometheusClient;
+import static io.trino.plugin.prometheus.PrometheusQueryRunner.createPrometheusQueryRunner;
+import static io.trino.spi.connector.NotPartitionedPartitionHandle.NOT_PARTITIONED;
+import static java.util.concurrent.TimeUnit.DAYS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertEquals;
+
+public class TestPrometheusIntegration
+        extends AbstractTestQueryFramework
+{
+    private static final int NUMBER_MORE_THAN_EXPECTED_NUMBER_SPLITS = 100;
+
+    private PrometheusServer server;
+    private PrometheusClient client;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        this.server = closeAfterClass(new PrometheusServer());
+        this.client = createPrometheusClient(server);
+        return createPrometheusQueryRunner(server, ImmutableMap.of(), ImmutableMap.of());
+    }
+
+    @Test
+    public void testSelectTable()
+    {
+        MaterializedResult results = computeActual("SELECT * FROM prometheus.default.up LIMIT 1");
```

Comment:
Use `assertQuery()`.

Reply:
Done, use the _skippingTypesCheck_ , if NOT use _skippingTypesCheck_ , we will hit error
 ```
org.junit.ComparisonFailure: [Output types for query [SELECT labels FROM prometheus.default.up LIMIT 1]] expected:<[map(varchar[(8), varchar(14)])]> but was:<[map(varchar[, varchar])]>
```

Reply:
You can avoid the error by fixing like this
```java
        assertThat(query("SELECT labels FROM prometheus.default.up LIMIT 1"))
                .matches("SELECT MAP(ARRAY[VARCHAR 'instance', '__name__', 'job'], ARRAY[VARCHAR 'localhost:9090', 'up', 'prometheus'])");
```

Reply:
@ebyhr  Thanks for your suggestion! Your suggestion worked well, I tried on the local. Thanks again! Done the changes.